docs/source/user_guide/gcm_based_inference/introduction.rst

```diff
@@ -181,5 +170,69 @@ This intervention says: "I'll ignore any causal effects of X on Y, and set every
 to 2.34." So the distribution of X will remain unchanged, whereas values of Y will be at a fixed
 value and Z will respond according to its causal model.
 
+These are the basic steps that need to happen. While we can run these steps explicitly, often they get
+executed as part of other steps, e.g. when fitting and re-fitting as part of computing confidence
+intervals. The next section therefore dives into a more typical usage pattern of the ``dowhy.gcm`` package.
+
+Typical usage of the ``dowhy.gcm`` package
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In practice, we may not execute the steps we've learned above explicitly and in this order, but they
+get called by other abstractions. E.g. we usually want to use confidence intervals, when answering
+causal questions, to quantify our confidence in the results. In this case, ``fit`` will be called on
+behalf of us, but we won't call it directly.
+
+At the same time, we may introduce some optional steps, again to improve our confidence in the
+results, such as some basic checks to see whether the independence structure of the graph is
+consistent with the data.
+
+Modeling an SCM and validating the structure of its causal graph against the data
+---------------------------------------------------------------------------------
+
+The graph modeling is the same as in `Step 1: Modeling cause-effect relationships as a structural
+causal model (SCM)`_. First we construct the causal graph:
+
+>>> causal_model = gcm.StructuralCausalModel(nx.DiGraph([('X', 'Y'), ('Y', 'Z')])) # X → Y → Z
+
+Now, we test this graph against our data:
+
+>>> rejection_result, _ = gcm.refute_causal_structure(causal_model.graph, data)
+>>> if rejection_result == gcm.RejectionResult.REJECTED:
+>>>      print("Do not continue. We advice revising the causal graph.")
+
+.. TODO: Add: For more details on structure validation and how to treat its results, see chapter
+:doc:`validating_causal_structure`.
+
+Unless the rejection result is ``REJECTED``, we continue to assign causal models:
```

Comment:
I am a bit concerned that most structures will be `REJECTED` due to the statistical issues with the independence tests, even if the structures are correct.

Reply:
Wait, but didn't we discuss this at length and said that `REJECTED` would only be returned when you really shouldn't continue, whereas in other cases it might still not be correct, but it's somewhat inconclusive. It feels like we're re-discussing something that we've already resolved, no?

Reply:
The discussion we had was to give a clear answer whether the graph is "correct", while we can only clearly say "no" if we `REJECTED` it. However, even if the graph is completely correct, there is a (very high) chance that we still return `REJECTED` here. At least, even in our artificial notebook examples, its often rejected although the data perfectly follows the graph structure.